spec/controllers/posts_controller_spec.rb

```diff
@@ -1052,6 +1052,85 @@
       end
     end
 
+    it "gives the correct previous post" do
+      user = create(:user)
+      board = create(:board, creator: user)
+      section = create(:board_section, board: board)
+      extra = create(:post, user: user, board: board, section: section)
+      prev = create(:post, user: user, board: board, section: section)
+      hidden = create(:post, board: board, section: section, privacy: Concealable::PRIVATE)
+      post = create(:post, user: user, board: board, section: section)
+      extra.update!(section_order: 0)
+      prev.update!(section_order: 1)
+      hidden.update!(section_order: 2)
+      post.update!(section_order: 3)
```

Comment:
Shouldn't the posts be in this order anyway? Maybe just check `expect([extra, prev, hidden, post].map(&:section_order)).to eq([0, 1, 2, 3])`